app/src/main/java/de/danoeh/antennapod/adapter/DataFolderAdapter.java

```diff
@@ -0,0 +1,137 @@
+package de.danoeh.antennapod.adapter;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RadioButton;
+import android.widget.TextView;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.util.Converter;
+import de.danoeh.antennapod.core.util.StorageUtils;
+import de.danoeh.antennapod.dialog.ChooseDataFolderDialog;
+import me.zhanghai.android.materialprogressbar.MaterialProgressBar;
+
+
+public class DataFolderAdapter extends RecyclerView.Adapter<DataFolderAdapter.ViewHolder> {
+    private final ChooseDataFolderDialog.RunnableWithString selectionHandler;
+    private final String currentPath;
+    private final List<StoragePath> entries;
+    private Dialog dialog;
+
+    public DataFolderAdapter(Context context, ChooseDataFolderDialog.RunnableWithString selectionHandler) {
+        this.entries = getStorageEntries(context);
+        this.currentPath = getCurrentPath();
+        this.selectionHandler = selectionHandler;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View entryView = inflater.inflate(R.layout.choose_data_folder_dialog_entry, parent, false);
+        return new ViewHolder(entryView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        StoragePath storagePath = entries.get(position);
+        holder.path.setText(storagePath.getShortPath());
+        holder.size.setText(Converter.byteToString(storagePath.getAvailableSpace()));
+        holder.progressBar.setProgress(storagePath.getUsagePercentage());
+        holder.root.setOnClickListener((View v) -> selectAndDismiss(storagePath));
+        holder.radioButton.setOnClickListener((View v) -> selectAndDismiss(storagePath));
+        if (storagePath.getFullPath().equals(currentPath)) {
+            holder.radioButton.toggle();
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        if (currentPath == null) return 0;
```

Comment:
Please prefer using brackets, even if it is a very short statement.

Reply:
Any preference on using brackets but keeping it in the same line? Or just reformat the code and leave statements in separate lines?

Reply:
Please use a new line - just like it is done in long blocks.